frontend/javascripts/oxalis/model/sagas/isosurface_saga.ts

```diff
@@ -824,33 +824,53 @@ function _getLoadChunksTasks(
                 const loader = getDracoLoader();
 
                 const errorsWithDetails = [];
-                for (let chunkIdx = 0; chunkIdx < chunks.length; chunkIdx++) {
-                  const chunk = chunks[chunkIdx];
-                  try {
-                    const dracoData = dataForChunks[chunkIdx];
-
-                    const geometry = yield* call(loader.decodeDracoFileAsync, dracoData);
-
-                    // Compute vertex normals to achieve smooth shading
-                    geometry.computeVertexNormals();
-
-                    yield* call(
-                      {
-                        context: segmentMeshController,
-                        fn: segmentMeshController.addIsosurfaceFromGeometry,
-                      },
-                      geometry,
-                      id,
-                      chunk.position,
-                      // Apply the scale from the segment info, which includes dataset scale and mag
-                      scale,
-                      lod,
-                      layerName,
-                    );
-                  } catch (error) {
-                    errorsWithDetails.push({ error, chunk });
+
+                const chunksWithData = chunks.map((chunk, idx) => ({
+                  ...chunk,
+                  data: dataForChunks[idx],
+                }));
+                // Group chunks by position and merge meshes in the same chunk to keep the number
+                // of objects in the scene low for better performance. Ideally, more mesh geometries
+                // would be merged, but the meshes in different chunks need to be translated differently.
+                const chunksGroupedByPosition = _.groupBy(chunksWithData, "position");
+                for (const chunksForPosition of Object.values(chunksGroupedByPosition)) {
+                  // All chunks in chunksForPosition have the same position
+                  const position = chunksForPosition[0].position;
+
+                  const bufferGeometries = [];
+                  for (let chunkIdx = 0; chunkIdx < chunksForPosition.length; chunkIdx++) {
+                    const chunk = chunksForPosition[chunkIdx];
+                    try {
+                      const bufferGeometry = yield* call(loader.decodeDracoFileAsync, chunk.data);
+                      bufferGeometries.push(bufferGeometry);
+                    } catch (error) {
+                      errorsWithDetails.push({ error, chunk });
+                    }
                   }
+
+                  const geometry = mergeBufferGeometries(bufferGeometries);
+
+                  // If mergeBufferGeometries does not succeed, the method logs the error to the console and returns null
```

Comment:
just to doublecheck: did you deliberately test this or did mergeBufferGeometries fail sometimes? (as in, there were problems with it?)

Reply:
This was mentioned in the documentation and I also saw it in the source code of the method. It never happened during testing.